Add announce feature via new channel controller

LukeBillington opened the pull request:
Hello there, I'm requesting to add a new channel controller with the following commands.
New commands:
+ `!channels` - Lists all available channels.
+ `!announce <channel_name>,[channel_name],[channel_name] message` - Broadcasts a message to one or more multiple channels.

LukeBillington committed:
Added channels controllers

This has 2 commands to start, !channels to list all channels and
!announce to broadcast to certain channels.

LukeBillington committed:
Modified help to include announce commands

Help in client.js updated with announce and channel commands.

LukeBillington committed:
Update usage docs

Added the new announce and and channels command to the usage document.

reactivepixel commented on bot/controllers/channels.js:
```diff
@@ -0,0 +1,111 @@
+        resType: 'dm',
+        action: (message) => {
+          const channels = [];
+          util.log('help!', message);
```
This log seems out of place. 

I suggest it's removal or enhancement to provide more contextual information with the title.

reactivepixel commented on bot/controllers/channels.js:
```diff
@@ -0,0 +1,111 @@
+            }
+            let msgParsedIndex = 2;
+            let preparedMessage = '';
+            while (msgParsedIndex < msg.parsed.length) {
```
Consider commenting this loop to add info on what is being done. Comments should make the source code skimmable for other devs.

LukeBillington committed:
Removed random log, commented loop

Removed a random log that output help. Commented the prepared message
loop that combines each term.

reactivepixel commented:
For final inclusion there needs to be an officer restriction on the role level of the user initiating, and also the bot should specify which user generated the announcement.
